### ISSUE TITLE
[feature] Better error description when installing packages with symlinks on Windows

### ISSUE THREAD

**Issue by sykhro** [conan-io/conan#14456]
### What is your suggestion?

To my knowledge, there is no guideline against using symlinks in conan packages.
Some projects create those by default. For example, LLVM creates a number of symlinks to some executables (e.g. clang-cl.exe is a symlink to clang.exe). What happens when you try to install such a package on Windows out of the box? This:
```
...\conan\internal\deploy.py", line 71, in full_deploy
    shutil.copytree(dep.package_folder, new_folder, symlinks=True)
  File "C:\Program Files\Python311\Lib\shutil.py", line 561, in copytree
    return _copytree(entries=entries, src=src, dst=dst, symlinks=symlinks,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python311\Lib\shutil.py", line 515, in _copytree
    raise Error(errors)
shutil.Error: [...] [WinError 1314] A required privilege is not held by the client:
```
By default only administrators can create symlinks on Windows. This causes `copytree()` to fail unless Conan is being run in an elevated context.

Should recipes ban symlinks in their folders? It would be nice to have a more descriptive error message for this occurrence, which I can't imagine being too rare.

For anyone arriving here from Google: **turn on "Developer Mode" on Windows**. It will fix this without having to edit group policies or running as admin.
You can check programmatically for this setting by checking if this `DWORD` is set to 1
```
HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\AppModelUnlock\AllowDevelopmentWithoutDevLicense
```

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @sykhro 

Thanks for your report.
This is related to the error message in https://github.com/conan-io/conan/issues/14451, thanks very much for the insight of enabling developer mode. 

I agree that at least improving the error message should be done. I am not sure what to check for, as there could be other reasons to make the copytree fail, checking the error msg is not robust, it could be localized, so we need to check this. At least, not causing a stack trace but a friendlier error should be done.

**Comment by sykhro** [comment]
> Hi @sykhro
> 
> Thanks for your report. This is related to the error message in #14451, thanks very much for the insight of enabling developer mode.
> 
> I agree that at least improving the error message should be done. I am not sure what to check for, as there could be other reasons to make the copytree fail, checking the error msg is not robust, it could be localized, so we need to check this. At least, not causing a stack trace but a friendlier error should be done.

As usual, thanks for the quick reply :) 
I think matching against "[WinError 1314]" might be stable as that's how shutil wraps all `OSError`s.
Also, it looks like `shutil.copy2` is aware of this and puts a file in place of the symlink depending on some parameters being set https://github.com/python/cpython/blob/0a7f48b9a8d54809f1e9272337aefe2444158e64/Lib/shutil.py#L451.

In my local script that wraps conan, I'm warning the user with this:
```python
import winreg
aReg = winreg.ConnectRegistry(None, winreg.HKEY_LOCAL_MACHINE)
aKey = winreg.OpenKey(aReg, r"SOFTWARE\Microsoft\Windows\CurrentVersion\AppModelUnlock")
val = winreg.QueryValueEx(aKey, "AllowDevelopmentWithoutDevLicense")[0]
if val != 1:
    logging.fatal("'Developer Mode' appears to be disabled")
```
not very useful in solving #14451 though.

**Comment by memsharded** [comment]
Proposing https://github.com/conan-io/conan/pull/14461, in case you want to run from source it, and give it a try.

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/14461 has been merged for next 2.0.10 release, but early feedback if possible still very welcome.

**Comment by sykhro** [comment]
Both the option and the error look good, it definitely points people in the right direction. I will try it out locally to see how big of a hit on disk space we'd take from not copying the symlinks (for the time being I enforced enabling developer mode)
